src/components/flow/routers/ticket/TicketRouterForm.test.ts

```diff
@@ -0,0 +1,43 @@
+import { composeComponentTestUtils, mock, setMock } from 'testUtils';
+import {
+  createOpenTicketNode,
+  createOpenTicketAction,
+  getRouterFormProps
+} from 'testUtils/assetCreators';
+import * as utils from 'utils';
+import TicketRouterForm from './TicketRouterForm';
+
+mock(utils, 'createUUID', utils.seededUUIDs());
+
+const routerNode = createOpenTicketNode(createOpenTicketAction());
+const { setup } = composeComponentTestUtils(TicketRouterForm, getRouterFormProps(routerNode));
+
+describe(TicketRouterForm.name, () => {
+  describe('render', () => {
+    it('should render', () => {
+      const { wrapper } = setup();
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('updates', () => {
+    it('should update and save', () => {
+      const components = setup(true, { updateRouter: setMock() });
+      const instance = components.instance as TicketRouterForm;
+
+      instance.handleSubjectUpdate('New subject');
+      expect(instance.state).toMatchSnapshot();
+      instance.getButtons().primary.onClick();
+      expect(components.props.updateRouter).toMatchCallSnapshot();
+    });
+
+    it('should cancel changes', () => {
+      const components = setup(true, { updateRouter: setMock() });
+      const instance = components.instance as TicketRouterForm;
+
+      instance.handleSubjectUpdate('New subject');
+      instance.getButtons().secondary.onClick();
+      expect(components.props.updateRouter).not.toBeCalled();
+    });
+  });
+});
```

Comment:
I wish I had mentioned this earlier, but we've been phasing on `composeComponentTestUtils` for the excellent and much clearer react testing library. A more current router test example would be [WaitRouterForm.test.tsx](https://github.com/nyaruka/floweditor/blob/master/src/components/flow/routers/wait/WaitRouterForm.test.tsx)

Main difference here, is that accessing `instance` and calling functions directly is considered no-bueno. Instead, interact as a user firing events and then checking what happened.

Reply:
When I added this it did feel dirty to have to go back and make some methods in the component public just to call them here. Will update..

Reply:
There's a ton more (and probably better) examples if you look at tests that are `test.tsx` files. Webhook and Random are a couple others. Some have a bit of both, been refactoring those a bit at a time when I touch them.